mediacloud/mediawords/util/test_colors.py

```diff
@@ -0,0 +1,86 @@
+from mediawords.test.test_database import TestDatabaseWithSchemaTestCase
+from mediawords.util.colors import *
+
+
+def test_hex_to_rgb():
+    assert hex_to_rgb('ff0000') == (255, 0, 0,)
+    assert hex_to_rgb('FFFFFF') == (255, 255, 255,)
+    assert hex_to_rgb('#ff0000') == (255, 0, 0,)
+    assert hex_to_rgb('#FFFFFF') == (255, 255, 255,)
+
+
+def test_rgb_to_hex():
+    assert rgb_to_hex(255, 0, 0).lower() == 'ff0000'
+    assert rgb_to_hex(0, 0, 0).lower() == '000000'
+    assert rgb_to_hex(255, 255, 255).lower() == 'ffffff'
+
+
+def test_analogous_color():
+    starting_color = '0000ff'
+    colors = analogous_color(color=starting_color, my_slices=255, slices=255)
+    assert len(colors) == 256
+    assert colors[0].lower() == starting_color
+    assert colors[1].lower() == '0400ff'
+    assert colors[-2].lower() == '0008ff'
+    assert colors[-1].lower() == starting_color
+
+
+class TestGetConsistentColorTestCase(TestDatabaseWithSchemaTestCase):
+
+    def test_get_consistent_color(self):
+        partisan_colors = {
+            'partisan_2012_conservative': 'c10032',
+            'partisan_2012_liberal': '00519b',
+            'partisan_2012_libertarian': '009543',
+        }
+
+        for color_id, color in partisan_colors.items():
+            got_color = get_consistent_color(db=self.db(), item_set='partisan_code', item_id=color_id)
+            assert got_color == color
```

Comment:
how did this know what color to assign these?

Reply:
It appears that those colors were prefilled in [`mediawords.sql`](https://github.com/berkmancenter/mediacloud/blob/master/schema/mediawords.sql#L673-L676) when originally creating a schema. Added a comment that mentions that.